lib/scratch-buffers.c

```diff
@@ -299,9 +299,11 @@ scratch_buffers_register_stats(void)
   StatsClusterKey sc_key;
 
   stats_lock();
-  stats_cluster_logpipe_key_legacy_set(&sc_key, SCS_GLOBAL, "scratch_buffers_count", NULL);
+  stats_cluster_logpipe_key_set(&sc_key, "scratch_buffers_count", NULL, 0);
+  stats_cluster_logpipe_key_add_legacy_alias(&sc_key, SCS_GLOBAL, "scratch_buffers_count", NULL);
   stats_register_counter(0, &sc_key, SC_TYPE_QUEUED, &stats_scratch_buffers_count);
-  stats_cluster_logpipe_key_legacy_set(&sc_key, SCS_GLOBAL, "scratch_buffers_bytes", NULL);
+  stats_cluster_logpipe_key_set(&sc_key, "scratch_buffers_bytes", NULL, 0);
+  stats_cluster_logpipe_key_add_legacy_alias(&sc_key, SCS_GLOBAL, "scratch_buffers_bytes", NULL);
```

Comment:
[optional]

`queued` seems odd here.

Adding `stats_cluster_single_key_add_legacy_alias_with_name()` and converting this to a single counter would probably solve this issue (see my poc patch), but as this is an experimental feature and scratch buffers counters are not top prio ATM, I think it is okay like this for now.

[scratch-buffers.diff.txt](https://github.com/syslog-ng/syslog-ng/files/10783456/scratch-buffers.diff.txt)

Reply:
Thank you very much for the patch.